### DISCUSSION TITLE
MooseFS as Block Storage

### DISCUSSION THREAD

**Discussion by anon314159** [moosefs/moosefs#615] [Q&A]
Hello,

There's a few practical scenarios I would like to validate are viable regarding the use of sparse files backed by MooseFS with the primary goals of reducing metadata footprint, resolve the lots of small files problem and provide SAN like functionality with the use of iSCSI, NVME-oF or loopback devices. 

Scenario One:
Exporting loopback devices via SMB/NFS.  This involves exporting a volume, mounting the export via MooseFS fuse client, creating a sparse file of arbitrary size using DD or truncate, formating this file as XFS, mounting that file via loop device, and then sharing this file system via Samba or NFS.

Scenario Two:
Exporting a raw sparse file or loop device backed by MooseFS via iSCSI, or NVME-oF as a lun, that can then be served out to any Windows or Linux iSCSI initiator/client.  

I have conducted several experiments with the above scenarios within one of my development environments with pretty decent results but would like to know whether or not these use cases are fully supported by MooseFS.  While projects like XCP-NG supports VM's backed by MooseFS, I am not sure if these use cases are kind of taboo or potentially dangerous in terms of data integrity.

**Comment by tianon** [comment]
You might find `mfsbdev` interesting: https://moosefs.com/manpages/mfsbdev

(See also #575)

**Reply by anon314159** [reply]
> You might find `mfsbdev` interesting: https://moosefs.com/manpages/mfsbdev
> 
> (See also #575)

Awesome, thanks for the reference. You wouldn't happen to know the maximum supported size for each block file (assuming the target file system is XFS)?  Using block devices with mfsbdev seems like an excellent way to fan-out I/0 across all of the chunkservers while reducing the metadata footprint.
